### DISCUSSION TITLE
Sending commands to the Battery?

### DISCUSSION THREAD

**Discussion by esanya** [mtrossbach/noah-mqtt#10] [Ideas]
Hi Markus,

this is really great projekt!

What do you think? Is is possible to implement sending configuration to the battery (change operation modes of the battery, standard power output, etc) via your SW?

are you somehow intercepting the http calls made by the APP?

Thanks,
Regards,
Sandor

**Comment by mtrossbach** [comment]
Hi Sandor,

Thank you for your kind words and interest in the project!

Currently, the software does support setting the default system output power and configuring charging/discharge limits, so those aspects of battery management are indeed possible through the existing implementation.

To do this send a MQTT message to `noah2mqtt/{{your-device-serial}}/parameters/set`. The payload must be a json object, containing the one or more of the following properties:
- `output_power_w`: Setting the system output power in Watts (minimum 0, maximum 800)
- `charging_limit`: Setting to upper charging limit in percent (minimum 70, maximum 100)
- `discharge_limit`: Setting to lower discharge limit in percent (minimum 0, maximum 30)

If you want to change the upper charging or lower discharge limit you have to include both, `charging_limit`and `discharge_limit` because the Growatt API currently only supports setting both at the same time. If you only want to set the system output power, it is ok to only send the `output_power_w` property, e.g.:

```json
{
   "output_power_w": 500
}
```

Keep im mind, if you have set a schedule, it will always overwrite the default system output power.

However, setting the operation mode of the battery is a bit more complex. As of now, it's only possible to set the operation mode by configuring a schedule, but unfortunately, the feature to set schedules is not yet supported in the software.

Regarding your question about intercepting HTTP calls made by the app, yes, that is one of the methods used to understand and replicate the communication between the app and the battery system.

I hope this answers your questions! If you have any more suggestions or need further assistance, feel free to ask.

Best regards,
Markus

**Reply by esanya** [reply]
Thank you!

**Reply by jofranz** [reply]
Thanks @mtrossbach! Currently the mobile app seem to only support 50 watt power steps. From 0 or 50 to 800. Could you confirm if you can also set smaller steps like 23 watts using your project?

**Comment by umeissen** [comment]
I also wanted to thank you for the great software. 

Picking up the discussion from a year ago. Is there any plan for the future to implement the setting of operation mode using schedules?

**Comment by FHEMUser** [comment]
One idea :
The Shelly 3em used an other mode.
It called "smart controll" or similar.
If we can use this mode with mqtt we will be able to set the current power without changing the standard output settings.
If Growatt have (again) a problem I think the standard output setting will used instead of the values from the Shelly.

But I think this mode is only available if you have an Shelly set-up in the App ?

**Reply by xdatravelbug** [reply]
I second that: There is some rumor going around that the smart mode (Grid first?) uses RAM instead of flash/eeprom (whatever the ESP32 has implemented). That avoids wear-out of the memory cells, which have a certain max. write-lifecycle.

Anyway, the colleagues over @ [https://github.com/robertzaage/GroBro/blob/main/grobro/growatt_noah_registers.json](https://github.com/robertzaage/GroBro/blob/main/grobro/growatt_noah_registers.json) are currently working on identifying the registers if and how these can help to config the Noah2000 via home automation. They are talking about register 8 for the mode, but in my case I see the serial in register 8...

Where I am heading to is, if maybe the mode can be changed/forced via some setting. Meaning that we do not need the app and/or a real Shelly but can also use the powermeter IR smart reader data from, e.g., FHEM and forward that to the Noah2000.

**Reply by FHEMUser** [reply]
Oh this is nice.
But is this meaning at the moment all data send to the Noah will be written into flash or eeprom ?

I hope not! This will damage the Noah in 10 Months (20h a day, every 20 Seconds)

Edit:
Take a look into grobro.py
There is a table with registers. You see 8 is serial and a lot unknown.

I tried sending the work mode with the Noah mqtt application, but it do nothing. 

At the moment I reduce the interval to 30 seconds for writing new data to the Noah.

**Reply by xdatravelbug** [reply]
If you talk about the excel-table - that was coming from me :-)
I started it out of curiosity to check which, values/data I can get from the Noah2000.
You can do it yourself for your equipment @ server.growatt.com, but it is a manual process - one register after another :-(

**Reply by FHEMUser** [reply]
No not the Excel table.
In the file grobro.py there are a lot of parameters. 
8 is listed as serial, and other was showing as unknown.

**Reply by xdatravelbug** [reply]
Ah, ok, I was reading through file "growatt_noah_registers.json" and there is some mismatch to my values/data
